src/packages/@app/pages/EnvironmentExplorer/EnvironmentExplorerPage.ViewModel.ts

```diff
@@ -78,33 +78,88 @@ export class EnvironmentExplorerViewModel {
    */
   public updateVariables = async (_variable: any) => {
     const progressiveTab = createDeepCopy(this._tab.getValue());
-    progressiveTab.isSave = false;
     progressiveTab.variable = _variable;
+    progressiveTab.property.environment.variable = _variable;
     this.tab = progressiveTab;
-    this.environmentTabRepository.updateEnvironmentTab(
-      progressiveTab.id,
-      progressiveTab,
-    );
+    await this.tabRepository.updateTab(progressiveTab.tabId, progressiveTab);
+    this.compareRequestWithServer();
   };
 
   /**
-   * @description - updates environment properties
-   * @param _data - new environment properties
+  * Compares the current request tab with the server version and updates the saved status accordingly.
+  * This method is debounced to reduce the number of server requests.
+  * @return A promise that resolves when the comparison is complete.
+  */
+  private compareRequestWithServerDebounced = async () => {
+    let result = true;
+    const progressiveTab = createDeepCopy(this._tab.getValue());
```

Comment:
rename function name to `compareEnvironmentWithServerDebounced`

Reply:
Resolved